src/angularjs-currency-input-mask.spec.js

```diff
@@ -1,13 +1,14 @@
 describe('Formatting numbers',function() {
     beforeEach(module('cur.$mask'));
 
-    var $compile, $rootScope, $filter;
+    var $compile, $rootScope, $filter, $timeout;
 
-    beforeEach(inject(function(_$compile_, _$rootScope_, _$filter_){
+    beforeEach(inject(function(_$compile_, _$rootScope_, _$filter_, _$timeout_){
         // The injector unwraps the underscores (_) from around the parameter names when matching
         $compile = _$compile_;
         $rootScope = _$rootScope_;
         $filter = _$filter_;
+        $timeout = _$timeout_;
```

Comment:
- [ ] remove $timeout variable, and useless usage in test

Reply:
This $timeout is required, to trigger  the setCaretPosition.

> setCaretPosition: function(input, pos) {
>                     return $q(function(resolve, reject) {
>                         $timeout(function() {

If you restore the initial "input.focus();", the test does not fail without the $timeout.